hammr/commands/image/image.py

```diff
@@ -228,73 +228,34 @@ def do_deploy(self, args):
             if pimage == 2:
                 return 2
 
-            image_id = generics_utils.extract_id(pimage.imageUri)
-            if image_id is None or image_id == "":
-                printer.out("Image not found", printer.ERROR)
+            target_platform = ""
+            if pimage.targetFormat:
+                target_platform = pimage.targetFormat.name
+
+            if not is_targeted_cloud_compatible(pimage):
+                printer.out("Hammr only supports deployments for Amazon AWS and OpenStack.", printer.ERROR)
```

Comment:
I wonder if you can not remove this compatiblity test and add the error message at the end of this method because there is a test to see if the deployment is Amazon or OpenStack to use the right launch method.